parsons/action_kit/action_kit.py

```diff
@@ -1307,15 +1308,30 @@ def collect_upload_errors(self, result_array):
         for res in result_array:
             upload_id = res.get("id")
             if upload_id:
+                # Pend until upload is complete
                 while True:
                     upload = self._base_get(endpoint="upload", entity_id=upload_id)
-                    if not upload or upload.get("status") != "new":
+                    if upload.get("is_completed"):
                         break
                     else:
                         time.sleep(1)
-                error_data = self._base_get(
-                    endpoint="uploaderror", params={"upload": upload_id}
-                )
-                logger.debug(f"error collect result: {error_data}")
-                errors.extend(error_data.get("objects") or [])
+
+                # ActionKit limits length of error list returned
+                # Iterate until all errors are gathered
+                error_count = upload.get("has_errors")
+                limit = 20
+
+                error_pages = math.ceil(error_count / limit)
+                for page in range(0, error_pages):
+                    error_data = self._base_get(
+                        endpoint="uploaderror",
+                        params={
+                            "upload": upload_id,
+                            "_limit": limit,
+                            "_offset": page * limit,
+                        },
+                    )
+                    logger.debug(f"error collect result: {error_data}")
+                    errors.extend(error_data.get("objects") or [])
```

Comment:
I'm not sure why we need the `or []` here, but you could also do
`error_data.get("objects", [])` in the case where objects isn't defined.

Reply:
Oh, oops! I'm sorry, I saw the approval but not the comment. Let me think about this for a minute.

Reply:
It looks like `extend(error_data.get("objects") or [])` and `extend(error_data.get("objects", []))` are functionally the same. The `or` paradigm can be used to assign the first nonempty object in a set. Basically, we will extend `errors` by `error_data.get("objects")` if it is not `None`, or "extend" by `[]` otherwise. Your new code does the same thing. I'm more familiar with your way of doing this, though, so I'll make this suggested change! Thank you!